pydoctor/templatewriter/__init__.py

```diff
@@ -5,6 +5,364 @@
 <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
           "DTD/xhtml1-strict.dtd">
 '''
+from typing import Iterable, List, Optional, Dict, overload
+try:
+    from typing import Protocol, runtime_checkable
+except ImportError:
+    from typing_extensions import Protocol, runtime_checkable
+import abc
+from pathlib import Path
+import warnings
+import copy
+from  xml.dom import minidom
+
+from zope.interface import implementer, verify
+
+from twisted.web.iweb import ITemplateLoader
+from twisted.web.template import XMLString, Element
+
+from pydoctor.model import System, Documentable
+
+def parse_dom(text: str) -> minidom.Document:
+    """
+    Create a L{minidom} representaton of the XML string. 
+    """
+    try:
+        dom = minidom.parseString(text)
+    except Exception as e:
+        raise ValueError(f"Can't parse XML from text '{text}'. XML documents can have only one root. ") from e
+    else:
+        return dom
+
+class UnsupportedTemplateVersion(Exception):
+    """Raised when custom template is designed for a newer version of pydoctor"""
+    pass
+
+@runtime_checkable
+class IWriter(Protocol):
+    """
+    Interface class for pydoctor output writer. 
+    """
+
+    @overload
+    def __init__(self, htmloutput: str) -> None: ...
+    @overload
+    def __init__(self, htmloutput: str, template_lookup: 'TemplateLookup') -> None: ...
+
+    def prepOutputDirectory(self) -> None:
+        """
+        Called first.
+        """
+        ... 
```

Comment:
The docstring already counts as the mandatory statement for the function body, so the `...` is not needed here. Removing it would avoid it being counted as not-covered.